src/client/eppo-client-with-bandits.spec.ts

```diff
@@ -429,5 +430,129 @@ describe('EppoClient Bandits E2E test', () => {
         expect(mockLogBanditAction.mock.calls[1][0].actionProbability).toBeCloseTo(0.256);
       });
     });
+
+    describe('Assignment logging deduplication', () => {
+      let mockEvaluateFlag: jest.SpyInstance;
+      let mockEvaluateBandit: jest.SpyInstance;
+      let variationToReturn: string;
+      let actionToReturn: string | null;
+
+      // Convenience method for repeatedly making the exact same assignment call
+      function requestClientBanditAction(): Omit<IAssignmentDetails<string>, 'evaluationDetails'> {
+        return client.getBanditAction(flagKey, subjectKey, subjectAttributes, ['toyota', 'honda'], 'control');
+      }
+
+      beforeAll(() => {
+        mockEvaluateFlag = jest
+          .spyOn(Evaluator.prototype, 'evaluateFlag')
+          .mockImplementation(() => {
+            return {
+              flagKey,
+              subjectKey,
+              subjectAttributes,
+              allocationKey: 'mock-allocation',
+              variation: { key: variationToReturn, value: variationToReturn },
+              extraLogging: {},
+              doLog: true,
+              flagEvaluationDetails: {
+                flagEvaluationCode: 'MATCH',
+                flagEvaluationDescription: 'Mocked evaluation',
+              },
+            } as FlagEvaluation;
+          });
+
+        mockEvaluateBandit = jest
+          .spyOn(BanditEvaluator.prototype, 'evaluateBandit')
+          .mockImplementation(() => {
+            return {
+              flagKey,
+              subjectKey,
+              subjectAttributes: { numericAttributes: {}, categoricalAttributes: {} },
+              actionKey: actionToReturn,
```

Comment:
Where `actionToReturn` is injected as the assigned action for mocking `BanditEvaluator.evaluateBandit()`